presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -484,6 +498,11 @@ public PlanOptimizers(
                         new PushPartialAggregationThroughJoin(),
                         new PushPartialAggregationThroughExchange(metadata.getFunctionRegistry()),
                         new PruneJoinColumns())));
+        builder.add(new IterativeOptimizer(
```

Comment:
This rule uses cost model. Cost model loses estimates when there are any partial aggregation in between.